src/lib/interactionManager/types.ts

```diff
@@ -26,6 +26,28 @@ export enum FlowType {
   Authentication = 'Authentication',
   CredentialShare = 'CredentialShare',
   CredentialReceive = 'CredentialReceive',
+  Authorization = 'Authorization',
+}
+
+export enum AuthorizationType {
+  AuthorizationRequest = 'authorizationRequest',
+  AuthorizationResponse = 'authorizationResponse',
+}
+
+export interface AuthorizationResponse {
+  description: string
+  imageURL?: string
+  action?: string
+}
+
+export interface AuthorizationRequest extends AuthorizationResponse {
+  callbackURL: string
```

Comment:
This could be present in the response as well, like with most other interaction types (except for `Auth`).

Reply:
In that case, both the request and the response have the same type. Also it makes it more difficult to type guard. Is there a reason the `callbackURL` should be in the response as well?